src/sentry/preprod/models.py

```diff
@@ -0,0 +1,225 @@
+        """The upload is complete, but the artifact is not yet processed."""
+        PROCESSED = 3
+        """The artifact has been processed and is ready to be used."""
+        FAILED = 4
+        """The artifact failed to upload or process. Read the error_code and error_message for more details."""
+
+        @classmethod
+        def as_choices(cls):
+            return (
+                (cls.UPLOADING, "uploading"),
+                (cls.UPLOADED, "uploaded"),
+                (cls.PROCESSED, "processed"),
+                (cls.FAILED, "failed"),
+            )
+
+    class ArtifactType:
+        XCARCHIVE = 0
+        """Apple Xcode archive."""
+        AAB = 1
+        """Android App Bundle."""
+        APK = 2
+        """Android APK."""
+
+        @classmethod
+        def as_choices(cls):
+            return (
+                (cls.XCARCHIVE, "xcarchive"),
+                (cls.AAB, "aab"),
+                (cls.APK, "apk"),
+            )
+
+    class ErrorCode:
+        UNKNOWN = 0
+        """The error code is unknown. Try to use a descriptive error code if possible."""
+        UPLOAD_TIMEOUT = 1
+        """The upload timed out."""
+        ARTIFACT_PROCESSING_TIMEOUT = 2
+        """The artifact processing timed out."""
+        ARTIFACT_PROCESSING_ERROR = 3
+        """The artifact processing failed."""
+
+        @classmethod
+        def as_choices(cls):
+            return (
+                (cls.UNKNOWN, "unknown"),
+                (cls.UPLOAD_TIMEOUT, "upload_timeout"),
+                (cls.ARTIFACT_PROCESSING_TIMEOUT, "artifact_processing_timeout"),
+                (cls.ARTIFACT_PROCESSING_ERROR, "artifact_processing_error"),
+            )
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    # Having a FK to both Org/Project is unnecessary
+    organization_id = BoundedBigIntegerField(db_index=True)
+    project = FlexibleForeignKey("sentry.Project")
+
+    # Nullable in case the file upload fails
+    file = FlexibleForeignKey("sentry.File", null=True)
+
+    # The date the artifact was built. E.g. an artifact could be built on 05/21/2025,
+    # but the user uploaded it on 05/22/2025.
+    date_built = models.DateTimeField(null=True)
+
+    build_configuration = FlexibleForeignKey(
+        "sentry.PreprodBuildConfiguration", null=True, on_delete=models.SET_NULL
+    )
+
+    state = BoundedPositiveIntegerField(
+        default=ArtifactState.UPLOADING, choices=ArtifactState.as_choices()
+    )
+
+    artifact_type = BoundedPositiveIntegerField(choices=ArtifactType.as_choices(), null=True)
+
+    error_code = BoundedPositiveIntegerField(choices=ErrorCode.as_choices(), null=True)
+    error_message = models.TextField(null=True)
+
+    # E.g. 1.2.300
+    build_version = models.CharField(max_length=255, null=True)
+    # E.g. 9999
+    build_number = BoundedBigIntegerField(null=True)
+    # Some artifacts are embedded with a UUID, e.g. iOS Mach-O binaries
+    build_uuid = models.UUIDField(null=True)
```

Comment:
Just checking that this couldn't be a link to a sentry `Release`. I assume it doesn't map one to one, but wanted to be sure.

Reply:
Yea there likely will be at some point in the form of an nullable foreign key. We are planning to add that later once we play around with Releases more.